src/content/docs/patch-signing.mdx

```diff
@@ -0,0 +1,142 @@
+---
+title: Patch Signing (preview)
+description: How to sign patches
+---
+
+{/* cspell:words outform pubout */}
+
+:::note
+We've shipped this preview of patch signing to unblock certain customer
+evaluations. While we're confident that this preview adds a useful layer of
+security, it does not yet have all the expected integrations with cloud signing,
+keychains, etc. that a final product would expect to have. We also do not yet
+manage signing keys for you or support many key formats or signature types.
+Signed patches are safe and ready for use, but may have rough edges or missing
+conveniences. Any customer should feel welcome to use this preview in
+production, and is encouraged to file feedback for any rough edges you may find.
+:::
+
+In addition to our default security measures, Shorebird also provides optional
+patch signing.
+
+Patch signing allows developers to cryptographically sign patch updates with
+their own keys. This ensures that no one (including Shorebird) can change the
+content of your application without your private cryptographic keys.
+
+Signing works in two parts. First, `shorebird release` commands can take an
+optional `--public-key-path` argument to embed a your public key in your
+released application. The Shorebird updater will enforce that only patches
+signed with a corresponding private key will be allowed to load for applications
+that include a public key.
+
+Second, when you build your patch with `shorebird patch`, you can pass
+`--private-key-path` to have Shorebird sign your patch with your private key.
+This is required if you created your release with a public key included.
+
+There are no required changes to your code and you can add or remove this
+signing requirement at any time by simply making a new release of your
+application.
+
+## Adding patch signing to your application
+
+To start, you will need an RSA key pair. Shorebird has only tested with a
+limited set of signing algorithms at this time. Please contact us if you have
+other requirements—we'd be happy to work with you to support your needs.
+
+The `shorebird` tool expects to be able to read private and public keys
+from `.pem` files on disk. If you need other ways of accessing your key material
+(or signing via a cloud signing service) please contact us. We'd be happy to
```

Comment:
I feel this paragraph kinda steps into what the previous one already said, about we being happy to support other needs.